src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -622,4 +627,77 @@ private static int findSlotForItemStack(@NotNull final IInventory inventory, @No
 
         return -1;
     }
+
+    /**
+     * Verifies if there is one tool with an acceptable level
+     * in a worker's inventory.
+     *
+     * @param tool      the type of tool needed
+     * @param inventory the worker's inventory
+     * @param hutLevel  the worker's hut level
+     * @return true if tool is acceptable
+     */
+    public static boolean hasToolLevel(final String tool, @NotNull final InventoryCitizen inventory, final int hutLevel)
+    {
+        for (int i = 0; i < inventory.getSizeInventory(); i++)
+        {
+            final ItemStack item = inventory.getStackInSlot(i);
+            final int level = Utils.getMiningLevel(item, tool);
+
+            if (Utils.isTool(item, tool) && verifyToolLevel(item, level, hutLevel))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Verifies if an item has an appropriated grade.
+     *
+     * @param item     the type of tool needed
+     * @param level    the type of tool needed
+     * @param hutLevel the worker's hut level
+     * @return true if tool is acceptable
+     */
+    public static boolean verifyToolLevel(final ItemStack item, int level, final int hutLevel)
+    {
+        if (item == null || hutLevel > FREE_TOOL_CHOICE_LEVEL)
+        {
+            return true;
+        }
+        else if (item.hasEffect() && hutLevel <= EFFECT_TOOL_CHOICE_LEVEL)
+        {
+            return false;
+        }
+        else if (hutLevel >= level)
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Assigns a string containing the grade of the toolGrade.
+     *
+     * @param toolGrade the number of the grade of a tool
+     * @return a string corresponding to the tool
+     */
+    public static String swapToolGrade(final int toolGrade)
+    {
+        switch (toolGrade)
+        {
+            case 0:
+                return "Wood or Gold";
+            case 1:
+                return "Stone";
+            case 2:
+                return "Iron";
+            case 3:
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)